### PR TITLE
Some polishing of `sisl.nodes`

### PR DESCRIPTION
**Pull request by pfebrer**
Just some minor changes that I made to `sisl.nodes` while working on the GUI.

- Added a registry to keep track of created node classes
- Changed `getattr` of `Plot` so that plots are not computed undesirably.
- Minor change in `FileNode`
- Keep track of nodified modules so that they can be returned if the module needs is requested to be nodified again.

### CODE REVIEW

**Comment by pfebrer** [comment]
BTW I wonder if `pre-commit` could make sure that the license comments are at the beggining of the file when a new file is created :thinking:

File: src/sisl/nodes/file_nodes.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 except ImportError:
     WATCHDOG_IMPORTED = False
 
+import sisl
```
[the remark this answers is not in the document]
Comment (zerothi) [reply src/sisl/nodes/file_nodes.py:16]: should this be deleted?

**Comment by zerothi** [comment]
> BTW I wonder if `pre-commit` could make sure that the license comments are at the beggining of the file when a new file is created 🤔

yes, that would be cool.

File: src/sisl/nodes/file_nodes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pfebrer) [reply src/sisl/nodes/file_nodes.py:16]: Yes 😅

**Comment by pfebrer** [comment]
> Have you seen #712

No, I hadn't seen it. I'll try to fix it tomorrow

**Comment by pfebrer** [comment]
I added the fix to hide the test warnings in this PR, and in principle this is ready.

**Review by zerothi — APPROVED** [review]

**Pull request merged**